src/Microsoft.AspNetCore.WebSockets/Internal/HandshakeHelpers.cs

```diff
@@ -63,17 +64,19 @@ public static bool CheckSupportedWebSocketRequest(string method, IEnumerable<Key
             return validConnection && validUpgrade && validVersion && validKey;
         }
 
-        public static IEnumerable<KeyValuePair<string, string>> GenerateResponseHeaders(string key, string subProtocol)
+        public static void GenerateResponseHeaders(string key, string subProtocol, HttpContext context)
```

Comment:
You might as well reduce HttpContext here to HeaderCollection, you don't use anything else and it'd be cleaner.